codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -316,10 +332,10 @@ public Http2Stream setPriority(int parentStreamId, short weight, boolean exclusi
             // Already have a priority. Re-prioritize the stream.
             weight(weight);
 
-            if (newParent != parent() || exclusive) {
+            if (newParent != parent() || (exclusive && newParent.numChildren() != 1)) {
```

Comment:
@nmittler - Please verify.  We were created a circular dependency in the priority tree without this check.  I found it in one of the new unit tests with a StackOverFlow excpetion.